### ISSUE TITLE
Hot key for QR code generation/popup

### ISSUE THREAD

**Issue by rugk** [rugk/offline-qr-code#54]
It would be nice to be able to press Ctrl+R or so (preferably customizable) and get the popup open.

In such a case, it should be determinated, whether some text is selected. If it is, it should use that text for the QR code, otherwise just the URL.

---

If such a thing is possible…

**Comment by ENT8R** [comment]
Maybe https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/commands is what you were looking for...

**Comment by rugk** [comment]
Yeah, indeed. 

So maybe we can collect hot key combinations for different OSes that.

So as a default I would already propose. I just noticed "Ctrl+R" reloads the page, so we cannot use that. Ctrl+G also not available. The same with shift there, too…
Ctrl+Q quits the browser on many OSes (Linux and MacOS, I think), so we cannot use that too…

Mhh, which one is even still possible to use? :thinking: 

However, one thing will be clear: It is not customizable by the user. That makes things easier though, so I don't think this is particularly bad. :smile: 
The only thing we could do is offer some alternatives, but I think that is not really worth it. :smiley:

**Comment by rugk** [comment]
It seems changing the shortcut is actually possible to implement with Firefox, because [this add-on](https://addons.mozilla.org/de/firefox/addon/session-sync/) writes:
> Starting with Firefox 60 you can also change the default shortcut in the configuration panel (Firefox Beta, Firefox Developer Edition, and Firefox Nightly are already supported). The API is available only after Firefox 60.

So one would need to look at how they did it/what API they used.

**Comment by ENT8R** [comment]
The source code is at https://github.com/ReDEnergy/SessionSync

**Comment by rugk** [comment]
As for v1.2 I think a default hotkey would, however be enough…

So I've opened https://github.com/rugk/offline-qr-code/issues/67 for this as a follow-up issue.

**Comment by rugk** [comment]
I think the hardest thing about this issue here actually is not the hot key itself, but that:
> In such a case, it should be determinated, whether some text is selected. If it is, it should use that text for the QR code, otherwise just the URL.

This may require some content script to be injected (remember, @ENT8R :wink:; I said we might need this again… :laughing:). Of course, not doing so is always preferred, but I see no other way of accomplishing it without some injection.
